model/TaskLogBroker/RdsTaskLogBroker.php

```diff
@@ -225,6 +228,109 @@ public function getReport($taskId)
     }
 
     /**
+     * @inheritdoc
+     */
+    public function findAvailableByUser($userId)
+    {
+        try {
+            $filters = [
+                [
+                    'column' => 'status',
+                    'columnSqlTranslate' => ':status',
+                    'operator' => '!=',
+                    'value' => TaskLogInterface::STATUS_ARCHIVED
+                ],
+                [
+                    'column' => 'owner',
+                    'columnSqlTranslate' => ':owner',
+                    'operator' => '=',
+                    'value' => $userId
+                ],
+            ];
+
+            $qb = $this->getQueryBuilder()
+                ->select('*')
+                ->from($this->getTableName());
+
+            foreach ($filters as $filter) {
+                $qb->andWhere($filter['column'] . $filter['operator'] . $filter['columnSqlTranslate'])
+                    ->setParameter($filter['column'], $filter['value'])
+                ;
+            }
+
+            $rows = $qb->execute()->fetchAll();
+            $collection = TaskLogCollection::createFromArray($rows);
+
+        } catch (\Exception $exception) {
+            \common_Logger::w('Something went wrong getting task logs ' . $exception->getMessage());
+
+            $collection = TaskLogCollection::createEmptyCollection();
+        }
+
+        return $collection;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getByIdAndUser($taskId, $userId)
+    {
+        $qb = $this->getQueryBuilder()
+            ->select('*')
+            ->from($this->getTableName())
+            ->andWhere('id = :id')
+            ->setParameter('id', (string) $taskId)
+            ->andWhere('status != :status')
+            ->setParameter('status', (string) TaskLogInterface::STATUS_ARCHIVED)
+            ->andWhere('owner = :owner')
+            ->setParameter('owner', (string) $userId)
+        ;
+
+        $row = $qb->execute()->fetch();
+
+        if ($row === false) {
+            throw new \common_exception_NotFound('Entity not found');
+        }
+
+        return TaskLogEntity::createFromArray($row);
```

Comment:
The same logic here, Entity is something which related to ORM system for me and unfortunately there is no such thing in TAO so it can be a bit misleading and not consequent. So I would suggest to operate with basic data structures (like array) just to be stay consequent.

Reply:
We should stay consequent with the idea of passing arrays? Entity it's just an entity, an ORM system it's using this terminology but they did not reserve this only for ORM I think. I like the idea of an entity which should be able to create itself from array, from csv, from any other resource, so it's not coupled to the database at all, as an ORM is.